# tfs-ruby-3-4-x-rubygems-openssl-require.patch
#
# Target:  lib/rubygems/openssl.rb (pass1 (all platforms))
# Applies: ruby 3.4.x (3.4.1 - 3.4.2)
# Source:  tebako gem lib/tebako/packager/rubygems_patch.rb RubygemsPatch::RUBYGEMS_OPENSSL_RB_PATCH
#
# 'autoload :OpenSSL, "openssl"' fails to deal with a default gem from a
# statically linked extension; require openssl eagerly instead.
diff --git a/lib/rubygems/openssl.rb b/lib/rubygems/openssl.rb
index c44f619..2a4bb2a 100644
--- a/lib/rubygems/openssl.rb
+++ b/lib/rubygems/openssl.rb
@@ -1,7 +1,11 @@
 # frozen_string_literal: true
 
+# Start of tebako patch
+require "openssl"
+# End of tebako patch
 autoload :OpenSSL, "openssl"
 
+
 module Gem
   HAVE_OPENSSL = defined? OpenSSL::SSL # :nodoc:
 end
